scvi/module/_vae.py

```diff
@@ -166,6 +173,18 @@ def __init__(
                 "{}.format(self.dispersion)"
             )
 
+        self.batch_embedding = None
+        if use_batch_embedding:
+            if n_batch < 2:
+                raise ValueError(
+                    "`adata` must have more than 1 batch when passing in "
+                    "`use_batch_embedding=True`."
+                )
+
+            self.batch_embedding = nn.Embedding(n_batch, batch_embedding_dim)
+            # fclayers expects size of batch input == n_batch
+            n_batch = batch_embedding_dim
```

Comment:
Currently, categorical_covariates are not batch_embedded. I guess it's fine but depending on the use case it might be useful.

Reply:
Good point! Do you think supporting other categorical covariates would be an interesting use case? In terms of the implementation, it would make sense if we supported embedding other covariates, we would just have to think a bit more to make the API non-redundant.

Reply:
I think batch_key and categorical_covariate in the same low-dimensional batch_embedding are fine (simply adding up both embeddings). It would be good to keep the names separate though. I have to think more about the use case. It requires cases with a large number of categorical covariate values and I haven't used those cases.